src/Bicep.Core.UnitTests/Utils/SourceArchiveBuilder.cs

```diff
@@ -16,38 +20,55 @@ public class SourceArchiveBuilder
 #endif
 
         private string? cacheRoot = null;
-        private Uri entrypointUri = PathHelper.FilePathToFileUrl(Rooted("foo/bar/entrypoint.bicep"));
-        private string entrypointBicepContents = "metadata hi = 'This is the bicep entrypoint source file'";
+
+        // First will be entrypoint, must be a bicep file
+        private List<ISourceFile> SourceFiles = new();
+
+        private ISourceFile EntrypointFile => SourceFiles[0];
 
         public SourceArchiveBuilder WithCacheRoot(string cacheRoot)
         {
             this.cacheRoot = cacheRoot;
             return this;
         }
 
-        public SourceArchiveBuilder WithEntrypointFile(Uri entrypoint, string entrypointBicepContents)
+        public SourceArchiveBuilder WithBicepFile(Uri fileUri, string contents)
         {
-            this.entrypointUri = entrypoint;
-            this.entrypointBicepContents = entrypointBicepContents;
+            SourceFiles.Add(SourceFileFactory.CreateBicepFile(fileUri, contents));
             return this;
         }
 
-        public SourceArchiveBuilder WithEntrypointFile(string entrypoint, string entrypointBicepContents)
+        public SourceArchiveBuilder WithBicepFile(string entrypoint, string contents)
         {
-            this.entrypointUri = PathHelper.FilePathToFileUrl(Rooted(entrypoint));
-            this.entrypointBicepContents = entrypointBicepContents;
+            WithBicepFile(PathHelper.FilePathToFileUrl(Rooted(entrypoint)), contents);
             return this;
         }
 
         public SourceArchive Build()
         {
-            var stream = SourceArchive.PackSourcesIntoStream(
-                entrypointUri,
+            var stream = BuildStream();
+            return SourceArchive.UnpackFromStream(stream).UnwrapOrThrow();
+        }
+
+        public Stream BuildStream()
```

Comment:
Is there a particulat advantage to operate on Streams vs BinaryData objects?

Reply:
Again, simply predates your change.